### PR TITLE
Lets you actually aim fireball

### PR DESCRIPTION
**Pull request by Crazylemon64**
Fireball is now a hand-based spell, similar to disintegrate - this will let you actually aim it, instead of launching it straight into a wall on accident.

:cl:Crazylemon
tweak: Makes fireball a hand-based spell, so you can actually aim it like you can with the wand
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I dislike this, personally, and I'll be more than happy to explain why: this already exists.

The staff of fireballs behaves exactly like a gun does, up to, and including targeting. Regular fireball, however, travels in a *perfectly* straight line, making it ideal for suppressive fire over long distances for the same generalized area (something very difficult to do with pixel projectiles+guns).

**Comment by Crazylemon64** [comment]
Given that fireball can normally be cast only once a minute, it sounds difficult to use it for suppressive fire... That, and I do not see a staff of fireballs, only a wand, which has 8 charges at most. Should I add a staff of fireballs?

**Comment by HereticChurch** [comment]
Fireball spell has a recharge rate of 6 seconds, not a minute, and can even be used while cuffed, which can be handy. This'd make it so you can't do that, removing some of it's utility. However, a staff of fireball sounds phenomenal, and would likely fill this role, if given a sufficiently slowed firerate and reduced damage compared to fireball wands (Which is likely what Fox was referring to).

**Comment by Crazylemon64** [comment]
While testing, I found that the `charge_max` variables were how long the spell took to recharge in seconds - meaning that by default, fireball takes a minute to recharge, as its `charge_max` is 60. SHOULD I lower this to 6 seconds?

**Comment by Crazylemon64** [comment]
Oh my goodness, the `dumbfire` spelltype is SO CONVOLUTED

Anyways, if I create a new magic bullet class to cast spells on hit, I should be able to get this going nicely - it looks like the fireball had an `inflict_handler` that would deal 20 + 25 combined burn and brute, as well as a separate a separate `explosion` class spell that doesn't seem to be called either, but since fireball isn't instarekking people, it seems like it's not proccing. Should I fix this?

**Comment by Fox-McCloud** [comment]
current behavior+cooldowns+etc should all be maintained.

**Comment by Crazylemon64** [comment]
So rip out the broken behavior?

**Comment by Crazylemon64** [comment]
Silly buttons

**Comment by Fox-McCloud** [comment]
Not really sure what you're referring to---broken behavior?

**Comment by Regen1** [comment]
Yessss
Please

**Comment by Dumbdumn5** [comment]
> Yessss
Please

**Comment by FlattestGuitar** [comment]
Did my massive failure inspire a PR?

I am honored.

**Comment by Spacemanspark** [comment]
:+1:

**Comment by Crazylemon64** [comment]
<a href=https://github.com/ParadiseSS13/Paradise/blob/master/code/datums/spells/wizard.dm#L251-L263>This broken behavior - it looks like the coders originally wanted it to have both the inflict and explode effect</a>, but instead they just created a new spell to cause an explosion, just above there.

**Comment by Fox-McCloud** [comment]
I'd test to make sure those aren't magically being called somehow (I don't see how they possibly could be), then ensure that the fireball that the wizman fires is identical to the current non-aimed version...then this should be merge ready.

**Comment by Fox-McCloud** [comment]
@Crazylemon64 ?

**Comment by Crazylemon64** [comment]
@Fox-McCloud Got assignments to work on, I'll refine this once I've got a good chunk of open time

**Comment by TullyBurnalot** [comment]
I like this idea. Sort of the wizard casting a fireball on his palm, then throwing it at someone. As it stands, I always felt like the current Fireball casting system (for the spell, at least) was ridiculously finicky, at best.

**Comment by TheDZD** [comment]
Yeah, I have to agree. Despite it encroaching on the niche of the staff of fireballs, the fireball spell in its current form (outside of this PR) is one of the less useful spells, as it tends to be as much of a hazard for the wizard as it is for crewmembers.

**Comment by Crazylemon64** [comment]
@TheDZD What is this "Staff of Fireballs" everyone keeps on mentioning? I see only a wand.

**Comment by HereticChurch** [comment]
Quit confusing the poor Lemon!
They mean the wand of fireballs. There is no staff of fireballs.

File: code/modules/projectiles/guns/magic/hand.dm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	qdel(src)
+
+/obj/item/weapon/gun/magic/hand/fireball
+	name = "readied fireball"
+	desc = "Don't pick your nose."
+
+	// Should be close enough
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	projectile_type = "/obj/item/projectile/magic/fireball"
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/magic/hand.dm]: https://github.com/ParadiseSS13/Paradise/blob/master/code/datums/spells/wizard.dm#L249

^That's the only thing the projectile should do; cause a flamey explosion. The projectile you used here does 10 brute, 10 fire, *and* explosion damage, so it's a significant buff.

Comment (Crazylemon64) [inline code/modules/projectiles/guns/magic/hand.dm]: I am aware. I need to get it to cast the spell instead, and trigger collision in a 1-tile radius too

**Comment by Crazylemon64** [comment]
No time, either.

**Pull request closed without merging**